test/unit/core/selectors.ts

```diff
@@ -1,30 +1,275 @@
+
+      expect(request.language).to.eq(language);
+    });
+
+    it('should apply defaults', () => {
+      const config: any = {
+        autocomplete: {
+          defaults: { suggestions: { c: 'd' } },
+          overrides: {},
+        }
+      };
+
+      const request: any = Selectors.autocompleteSuggestionsRequest(config);
+
+      expect(request.c).to.eq('d');
+    });
+
+    it('should override defaults', () => {
+      const config: any = {
+        autocomplete: {
+          defaults: { suggestions: { c: 'd' } },
+          overrides: { suggestions: { a: 'b', c: 'd1' } },
+        }
+      };
+
+      const request: any = Selectors.autocompleteSuggestionsRequest(config);
+
+      expect(request.a).to.eq('b');
+      expect(request.c).to.eq('d1');
+    });
+  });
+
+  describe('autocompleteProductsRequest()', () => {
+    it('should create a products request', () => {
+      const area = 'myArea';
+      const language = 'en';
+      const productCount = 41;
+      const config: any = {
+        autocomplete: {
+          area,
+          language,
+          productCount,
+          defaults: {},
+          overrides: {},
+        }
+      };
+
+      const request = Selectors.autocompleteProductsRequest(config);
+
+      expect(request).to.eql({
+        area,
+        language,
+        numProducts: productCount
+      });
+    });
+
+    it('should apply overrides', () => {
+      const language = 'fr';
+      const config: any = {
+        autocomplete: {
+          language: 'en',
+          defaults: {},
+          overrides: { products: { language } },
+        }
+      };
+
+      const request: any = Selectors.autocompleteProductsRequest(config);
+
+      expect(request.language).to.eq(language);
+    });
+
+    it('should apply defaults', () => {
+      const config: any = {
+        autocomplete: {
+          defaults: { products: { c: 'd' } },
+          overrides: {},
+        }
+      };
+
+      const request: any = Selectors.autocompleteProductsRequest(config);
+
+      expect(request.c).to.eq('d');
+    });
+
+    it('should override defaults', () => {
+      const config: any = {
+        autocomplete: {
+          defaults: { products: { c: 'd' } },
+          overrides: { products: { a: 'b', c: 'd1' } },
+        }
+      };
+
+      const request: any = Selectors.autocompleteProductsRequest(config);
+
+      expect(request.a).to.eq('b');
+      expect(request.c).to.eq('d1');
+    });
+  });
+
+  describe('area()', () => {
+    it('should select the global area from state', () => {
+      const area = 'myArea';
+      const state: any = { data: { area } };
+
+      expect(Selectors.area(state)).to.eq(area);
+    });
+  });
+
+  describe('area()', () => {
+    it('should select the fields from state', () => {
+      const fields = ['a', 'b'];
+      const state: any = { data: { fields } };
+
+      expect(Selectors.fields(state)).to.eq(fields);
```

Comment:
I think unless we explicitly expect it to be a different object, it's always simpler (and faster) to check direct equality